api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -205,27 +208,56 @@ public String toString() {
   }
 
   public static class TimestampType extends PrimitiveType {
-    private static final TimestampType INSTANCE_WITH_ZONE = new TimestampType(true);
-    private static final TimestampType INSTANCE_WITHOUT_ZONE = new TimestampType(false);
+
+    private static final TimestampType INSTANCE_MICROS_WITH_ZONE =
+        new TimestampType(true, ChronoUnit.MICROS);
+    private static final TimestampType INSTANCE_MICROS_WITHOUT_ZONE =
+        new TimestampType(false, ChronoUnit.MICROS);
+    private static final TimestampType INSTANCE_NANOS_WITH_ZONE =
+        new TimestampType(true, ChronoUnit.NANOS);
+    private static final TimestampType INSTANCE_NANOS_WITHOUT_ZONE =
+        new TimestampType(false, ChronoUnit.NANOS);
 
     public static TimestampType withZone() {
```

Comment:
Should we mark these as deprecated? And suggest using `microsWithZone()` instead?

Reply:
Fixed in d7b28e4b3aaa0610ea49ba15b9f3e5a1862f9fac